tests/integration_tests/functional/test_error_code.py

```diff
@@ -0,0 +1,42 @@
+# Copyright 2019 Amazon.com, Inc. or its affiliates. All Rights Reserved.
+# SPDX-License-Identifier: Apache-2.0
+"""Tests scenarios for Firecracker signal handling."""
+
+import os
+import time
+import platform
+import pytest
+
+
+@pytest.mark.skipif(
+    platform.machine() == "x86_64",
+    reason="We only test with a kernel for aarch64"
+)
+def test_enosys_error_code(test_microvm_with_ssh):
+    """Test that ENOSYS error_code doesn't kill the microVM."""
+    test_microvm = test_microvm_with_ssh
+    test_microvm.spawn()
+
+    # We don't need to monitor the memory for this test.
+    test_microvm.memory_monitor = None
+
+    test_microvm.kernel_file = \
+        test_microvm.kernel_file[:-11] + "enosys_vmlinux.bin"
```

Comment:
we should also have a comment saying what this kernel does.
if we want to update this kernel further down the line, we should know how to re-add the specific faulty instruction

Reply:
The kernel used for this test is one randomly generated. Also how the error code is handled is not influenced in any way by the kernel version. In my opinion we should create a new issue and try to reproduce what @sandreim  suggested in this comment https://github.com/firecracker-microvm/firecracker/pull/2350#discussion_r543330663